campaigns/workflows.md

```diff
@@ -0,0 +1,345 @@
+suggested in the [Search & Discovery](#search--discovery) section,
+above.  This phase involves submitting the Notebook to a "repository"
+so that other users can "download" the Notebook for modification or
+import elements of the Notebook's "public API" into their own Notebook.
+
+Ideally, published Notebooks will be tested for regression as new
+versions of the software stack change, so that owners of the Notebook
+know that their Notebook will no-longer work with updated versions of
+the Notebook environment.
+
+Ideally, published Notebooks will be versioned and tied to a specific
+environment/kernel with pinned versions of the software used in the
+Notebook.  Ideally, the version of the Notebook and the version of the
+environment can be updated independently of each other.
+
+Ideally, published Notebooks will be published with metadata that
+contains (at a minimum) a description of what the Notebook does, a
+list of "public API" elements, the author's name and contact information
+(email), the version of the Notebook, the date and timestamp of the
+last change to the Notebook.  Ideally, this metadata will also contain
+information about who has "downloaded" the Notebook for modification,
+who has used which "public API" elements of the Notebook, how many
+times has the Notebook been "downloaded" and how many times has each
+"public API" element been accessed.
+
+#### Possible Projects
+
+- A Continuous Integration service to test Notebooks on HPC (and/or in
+  the cloud) that informs owners of the Notebook of failures with
+  given versions of the software in the Notebook kernel
+- A Notebook publication repository (e.g., a Gallery) that is searchable,
+  as proposed above, with versioning of the Notebook file and its
+  kernel.
+
+### Platform
+
+The *Platform* is the suite of software products and deployments that
+make scalable scientific workflows, as described above, possible.  The
+platform currently exists as a combination of JupyterHub, JupyterLab,
+Binder, Xarray, Dask, Intake, and peripheral packages in the Pangeo
+stack.
+
+Ideally, all of these utilities will work exactly the same way on
+any platform, cloud or HPC.  Ideally, the user should *not* need to
+understand how to set up the environment.  Ideally, the user should
+be able to get as quickly to scientific analysis as possible, with as
+many features or concerns about the platform being abstracted away as
+possible.
+
+#### Possible Projects
+
+- Binder on HPC, or something that looks and works like Binder on HPC.
+  Unlike the cloud version of Binder, the HPC version could require
+  authentication.
+- Dask clusters that span different cloud and HPC environments.  This
+  would allow users to analyze datasets located at different locations
+  from the same Notebook.  And if only reductions were directly
+  compared, it would facility the computation of the reductions on the
+  "server side" (i.e., next to the data) and then transport only the
+  reduced dataset.
```

Comment:
@kmpaul, I am curious.... What are your thoughts on an experimental dask-gateway deployment on Cheyenne/Casper? It appears that this an experiment worth looking into especially now that we have services like the object storage and campaign storage that are exclusively accessible from casper. I am envisioning a use case in which a user may want to take full advantage of both casper and cheyenne i.e. have a heterogeneous dask-gateway cluster with:
- a dask cluster running on Cheyenne: this cluster would have access to GLADE
- another dask cluster running on Casper: this cluster would have access to stratus.ucar.edu, and the campaign storage

P.S.: I am not familiar with the planned specs for NWSC-3 system. So, this experiment may be irrelevant in regards to Cheyenne's replacement system.

Reply:
@andersy005 I think this is a great idea.  I think this should be spec'd out.  I'll add this idea as a possible project idea.

Reply:
Added with latest commit